src/adapter/graphql/src/mutations/datasets_mut.rs

```diff
@@ -109,7 +109,11 @@ impl DatasetsMut {
     ) -> Result<CreateDatasetFromSnapshotResult> {
         let dataset_repo = from_catalog::<dyn domain::DatasetRepository>(ctx).unwrap();
 
-        let result = match dataset_repo.create_dataset_from_snapshot(snapshot).await {
+        // TODO: read param from input
```

Comment:
you can create an optional parameter (defaulting to private) for compatibility

consider also using an input struct here to anticipate many other creation options to come

Reply:
Since all the tests add the in-memory version of the service, I decided to make the default version with `PubliclyAvailable` value.

What do you think?

Reply:
Not actual: Private is now used by default